enhancements/microshift/lvms-topolvm-downstream-replacement.md

```diff
@@ -0,0 +1,269 @@
+in as many places as possible to ensure that it is well tested and
+maintained.
+
+### Goals
+
+* Switch from using repackaged registry.redhat.io/lvms4/topolvm-rhel9 images
+  to using a minified version of LVMS in MicroShift based on registry.redhat.io/lvms4/lvms4-lvms-operator-bundle.
+* Switch from using CSI Sidecars to using the LVMS Deployment optimized for Edge.
+
+### Non-Goals
+
+* Ensuring that LVMCluster works for all use cases in MicroShift.
+  Instead, we will focus on allowing custom deviceClasses to be used 
+  outside the LVMCluster Custom Resource.
+* Integrating all usual MicroShift TopoLVM use cases into LVMS. (think of LVM RAID, etc.)
+  Instead, we will rebrand our documentation and put forward best practices 
+  for using LVMS in MicroShift based on LVMCluster with an extended documentation
+  for specialised use cases.
+* Forcing MicroShift users to migrate to LVMS CustomResources. We will allow
+  the use of the old deviceClass mechanism without a migration path and permanently
+  allow custom deviceClass management.
+
+## Proposal
+
+With 4.16 and the [controller consolitation initative in LVMS](https://issues.redhat.com/browse/OCPEDGE-689)
+we no longer use any of our own upstream images like TopoLVM or the CSI Sidecars. 
+We would like to introduce a conformant way to rebase MicroShift based on LVMS, not TopoLVM in the future.
+This will allow us to no longer ship topolvm as a component,
+but embed it into LVMS fully as well as maintain LVMS as the only
+storage driver in MicroShift and OpenShift we have to support for lvm2 support.
+
+### Workflow Description
+
+1. Edge device administrator deploys a host with MicroShift 4.15
+   installed, and maintains a storageClass, and deviceClass as per 
+   [the documentation reference](https://github.com/openshift/microshift/blob/main/docs/contributor/storage/configuration.md).
+2. Software runs, time passes.
+3. Edge device administrator updates the host to run MicroShift 4.16.
+   * For ostree-based systems, the host is automatically rebooted as
+     part of the update process.
+   * For RPM-based systems, the user must reboot the host after the
+     software update is completed.
+4. Edge device restarts.
+5. MicroShift restarts.
+6. MicroShift removes the old TopoLVM deployment without removing the CRDs.
+   All existing storageClasses and deviceClasses are still present.
+7. MicroShift installs LVMS with a LVMCluster CustomResource.
+   * The LVMCluster is configured with `.spec.configurationPolicy=ExternalConfiguration`
+     which will trigger a deployment of the driver which can use the existing
+     deviceClasses and storageClasses.
+   * If LVMS does not detect a lvmd.yaml file (at runtime), it will error out,
+     as an external configuration is now mandatory for correct configuration.
+     Note that if the lvmd.yaml file is not present at install time, we should
+     keep the current behavior of not installing LVMS at all.
```

Comment:
The storage driver installation is currently mandatory in MicroShift. Do we mean we'd like to keep the behavior of not `starting` LVMS when `lvmd.yaml` is not present?

Reply:
Here is my current plan that was also discussed in a separate thread with @dhellmann : https://github.com/openshift/enhancements/pull/1601#discussion_r1538816528 maybe thats relevant to this discussion?

Reply:
I (think that I) understand. I changed the wording now slightly to

> If LVMS does not detect a lvmd.yaml file (at runtime), it will error out,
     as an external configuration is now mandatory for correct configuration.
     Note that if the lvmd.yaml file is not present when starting MicroShift, we should
     [keep the current behavior of not enabling LVMS and skip deploying it from the assets.](https://github.com/openshift/microshift/blob/main/pkg/components/storage.go#L83)

Reply:
Resolving as I believe this was addressed